### ISSUE TITLE
Enable as a web extension?

### ISSUE THREAD

**Issue by whilp** [RandomFractals/vscode-data-preview#298]
It would be *amazing* to view and edit data files without a heavy installation. I imagine some of the dependencies here would make it hard to fully enable this extension for the web. But perhaps a useful subset could still be exposed?

https://code.visualstudio.com/api/extension-guides/web-extensions

**Comment by RandomFractals** [comment]
yeah, this extension is grossly out of date, and updating it to work with on https://vscode.dev might require some elbow grease. 

I am starting to work on a simpler tabular data viewer next week and will consider making it work in a browser too:

https://github.com/RandomFractals/tabular-data-viewer

**Comment by christian-bromann** [comment]
@RandomFractals I just went through this exercise in [Marquee](https://github.com/stateful/vscode-marquee/pull/28/files). Maybe this helps to get some inspirations. Main tasks were:

- create additional webpack bundle
- replace `fs` with `vscode.workspace.fs` where ever possible (and if not use `if (globalThis.process) { ... }` to exclude the code from the web extension bundle)
- load webview template via `vscode.workspace.fs` or bundle it with your extension

**Comment by RandomFractals** [comment]
@christian-bromann yes, I am familiar with the steps. I might do it eventually in this or the new tabular data viewer extension. I still think the number of devs using it in a web editor is a much smaller audience, and there are other simpler extensions that show data grids in the new web editor version.
